geode-core/src/main/java/org/apache/geode/cache/configuration/CacheConfig.java

```diff
@@ -1022,6 +1025,50 @@ public void setVersion(String value) {
     this.version = value;
   }
 
+  public RegionConfig findRegionConfiguration(String regionPath) {
```

Comment:
This class should only be a container for configuration objects.  Every addition to this class is another deviation from its expected behavior.

These are better off staying in `CacheElement`.  Alternatively, these could compose a small `CacheElementUtils` class.

These methods should be `static`.

Reply:
it just seems very intuitive to call cacheConfig.getRegonElement (id) or similar operations. It's counter-intuitive that developer has to know to call some class's static method to achieve this.